docs/userguide/miscellaneous.rst

```diff
@@ -4,11 +4,30 @@ Controlling files in the distribution
 =====================================
 
 For the most common use cases, ``setuptools`` will automatically find out which
-files are necessary for distributing the package.
-These include all :term:`pure Python modules <Pure Module>` in the
-``py_modules`` or ``packages`` configuration, and the C sources (but not C
-headers) listed as part of extensions when creating a :term:`source
-distribution (or "sdist")`.
+files are necessary for distributing the package. More precisely, the following
+files are included in a source distribution by default:
+
+- All Python source files implied by the ``py_modules`` and ``packages``
+  ``setup()`` arguments
+- All C source files mentioned in the ``ext_modules`` or ``libraries``
+  ``setup()`` arguments
+- Scripts specified by the ``scripts`` ``setup()`` argument
+- All files specified by the ``package_data`` and ``data_files``
+  ``setup()`` arguments
+- The file specified by the ``license_file`` option in ``setup.cfg``
+  (setuptools 40.8.0+)
```

Comment:
I think it is not worthy to keep track of all the versions directly here.
There will be cases that a parameter is supported since one version in `setup.py`, another version in `setup.cfg` and yet another version in `pyproject.toml`...

Maybe we can add a note at the end saying something like:

> Please note that the list above is guaranteed to work in the last stable version
> of ``setuptools``. The behaviour of older versions might differ.